### PR TITLE
Fix copy/paste when scribble enabled

### PR DESCRIPTION
**Pull request by justinmc**
The recent scribble PR (https://github.com/flutter/engine/pull/24224) broke cmd+c and cmd+v on scribble enabled devices.  The reason was [this line](https://github.com/flutter/engine/blob/main/shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm#L1051), which seems to have been added to avoid the "[app] pasted" notification from showing up on every keystroke.

This solution returns YES when it receives a real copy/paste action and not just a key input, so copy and paste work and you won't get the notification on normal keyboard input.

Fixes https://github.com/flutter/flutter/issues/98832

CC @fbcouch

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.mm [hunk 1]
```diff
@@ -866,6 +866,32 @@ - (void)testUITextInputAvoidUnnecessaryUndateEditingClientCalls {
   XCTAssertEqual(updateCount, 2);
 }
 
+- (void)testCanCopyPasteWithScribbleEnabled {
+  if (UI_USER_INTERFACE_IDIOM() != UIUserInterfaceIdiomPad) {
```
[the remark this answers is not in the document]
Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.mm:870]: I would suggest not making `IsScribbleAvailable` a static function. I don't see a particular reason `IsScribbleAvailable` has to be static.

**Comment by fbcouch** [comment]
I forget if I saw that "[app] pasted" notification. If I remember
correctly, I returned NO for copy paste there to prevent the native text
editing toolbar from showing up. As long as that is also not happening when
selecting text either with scribble or touch (or keyboard) then I think
this should be fine.

**Comment by justinmc** [comment]
I saw your comment about the native toolbar but I was never able to get it to appear while testing on the iPad.  I'll keep an eye out for it but hopefully we're good.

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.mm (hunk 1 above)
[the remark this answers is not in the document]
Comment (justinmc) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.mm]: Thanks for the suggestion, that worked!  This is now ready for full review.

**Review by cyanglaz — APPROVED** [review]
LGTM! Thanks for the fix!

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 2]
```diff
@@ -1048,7 +1048,7 @@ - (BOOL)canBecomeFirstResponder {
 - (BOOL)canPerformAction:(SEL)action withSender:(id)sender {
   // When scribble is available, the FlutterTextInputView will display the native toolbar unless
   // these text editing actions are disabled.
-  if (IsScribbleAvailable()) {
+  if ([self isScribbleAvailable] && sender == NULL) {
```
Comment (hellohuanlin) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm:1051]: @justinmc do you remember what does `sender == NULL` mean here? 

Also can you explain more about what this is trying to do? When I remove this `sender == NULL`, CMD+C/V from hardware keyboard still works. So I may be missing somehting. Thanks!

Comment (justinmc) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm:1051]: Sorry I didn't document this better, I'm struggling to remember...

Are you trying this on an iPad with a hardware keyboard attached?  And iOS 14+.

The relevant part of the [docs for canPerformAction](https://developer.apple.com/documentation/uikit/uiresponder/1621105-canperformaction) say:

 > The object calling this method. For the editing menu commands, this is the shared [UIApplication](https://developer.apple.com/documentation/uikit/uiapplication) object.

Looking at the original scribble PR (https://github.com/flutter/engine/pull/24224), we seem to call `performAction` a bunch.  Maybe it's called with no `sender` from some of those and/or when hardware keys are being pressed?

Are you considering removing that expression for some reason?  I would test a physical iPad and Apple Pencil and hardware keyboard before at least.

Comment (hellohuanlin) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm:1051]: I am not doing anything particular for this code. I was just looking into "secure paste" feature, and trying to understand the codebase.

Comment (hellohuanlin) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm:1051]: interestingly i found that this function is not even called when we copy/paste. The other function in `FlutterPlatformPlugin.mm` is called instead. This code seems to be obsolete

Comment (justinmc) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm:1051]: Nice, I'm happy to review if you want to remove it.

**Pull request merged**